sklearn/decomposition/tests/test_fastica.py

```diff
@@ -12,10 +12,18 @@
 from sklearn.utils._testing import assert_array_equal
 from sklearn.utils._testing import assert_allclose
 
-from sklearn.decomposition import FastICA, fastica, PCA
+from sklearn.decomposition import fastica, PCA
+from sklearn.decomposition import FastICA
+
 from sklearn.decomposition._fastica import _gs_decorrelation
 from sklearn.exceptions import ConvergenceWarning
 
+msg = (
+    r"From version 1.4 `whiten_solver='auto'` will be used by default. Manually set the"
+    r" value of `whiten_solver` to suppress this message."
+)
+pytestmark = pytest.mark.filterwarnings(f"ignore:{msg}:FutureWarning")
```

Comment:
Actually, it could be nice to set everywhere `solver="auto"` to be sure that the tests are indeed all working as expected and to not have failing tests at the time of changing the default.

This is a bit more tedious but at least it ensures it.